app/src/main/java/com/github/onlynotesswent/ui/overview/NoteOverview.kt

```diff
@@ -0,0 +1,139 @@
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.github.onlynotesswent.MainActivity
+import com.github.onlynotesswent.R
+import com.github.onlynotesswent.model.folder.Folder
+import com.github.onlynotesswent.model.folder.FolderViewModel
+import com.github.onlynotesswent.model.note.Note
+import com.github.onlynotesswent.model.note.NoteViewModel
+import com.github.onlynotesswent.model.user.UserViewModel
+import com.github.onlynotesswent.ui.common.FolderDialog
+import com.github.onlynotesswent.ui.common.NoteDialog
+import com.github.onlynotesswent.ui.navigation.BottomNavigationMenu
+import com.github.onlynotesswent.ui.navigation.LIST_TOP_LEVEL_DESTINATION
+import com.github.onlynotesswent.ui.navigation.NavigationActions
+import com.github.onlynotesswent.ui.navigation.Screen
+import com.google.firebase.Timestamp
+
+/**
+ * Displays the overview screen which contains a list of the user's notes and folders. The user can
+ * create new notes and folders from this screen. If there are no notes or folders, it shows a text
+ * to the user indicating that there are no notes or folders.
+ *
+ * @param navigationActions The navigationActions instance used to transition between different
+ *   screens.
+ * @param noteViewModel The ViewModel that provides the list of publicNotes to display.
+ * @param userViewModel The ViewModel that provides the current user.
+ * @param folderViewModel The ViewModel that provides the list of folders to display.
+ */
+@Composable
+fun NoteOverviewScreen(
+    navigationActions: NavigationActions,
+    noteViewModel: NoteViewModel,
+    userViewModel: UserViewModel,
+    folderViewModel: FolderViewModel
+) {
+  val userRootNotes = noteViewModel.userRootNotes.collectAsState()
+  val userRootFolders = folderViewModel.userRootFolders.collectAsState()
+
+  val parentFolderId = folderViewModel.parentFolderId.collectAsState()
+  val context = LocalContext.current
+
+  var expanded by remember { mutableStateOf(false) }
+  var showCreateFolderDialog by remember { mutableStateOf(false) }
+  var showCreateNoteDialog by remember { mutableStateOf(false) }
+
+  // Handle back press
+  BackHandler {
+    // Move the app to background
+    (context as MainActivity).moveTaskToBack(true)
+  }
+
+  Scaffold(
+      modifier = Modifier.testTag("overviewScreen"),
+      floatingActionButton = {
+        Column(
+            verticalArrangement = Arrangement.spacedBy(16.dp), // Space between FABs
```

Comment:
We should stop using magic numbers for padding and write a comprehensive guide on the amount of padding used throughout the app but it is outside of the scope of this PR

Reply:
yes 100% agreed. Really unfortunate that we don't have more time.